File: components/rust/example/app1/Sconscript.py
Import('rtconfig')
from building import *
import os
import subprocess

cwd = GetCurrentDir()

# 设置Rust编译环境变量
rust_env = os.environ.copy()
rust_env['RTT_PATH'] = Dir('#').abspath

# Rust编译函数
def build_rust_lib(target, source, env):
    print("Building Rust application...")
    os.chdir(cwd)
    
    # 执行cargo build
    result = subprocess.run([
        'cargo', 'build', 
        '--target', 'riscv64gc-unknown-none-elf',
        '--release'
    ], env=rust_env)
    
    if result.returncode != 0:
        raise Exception("Rust compilation failed")
    
    print("Rust build completed successfully")

# 创建Rust库目标
rust_target = os.path.join(cwd, 'target/riscv64gc-unknown-none-elf/release/librust_example.a')

# 注册构建任务
rust_lib = env.Command(
    rust_target,
    Glob('src/*.rs') + ['Cargo.toml'],  # 依赖源文件
    build_rust_lib
)

# 设置库路径和库名
LIBPATH = [os.path.join(cwd, 'target/riscv64gc-unknown-none-elf/release')]
LIBS = ['rust_example']

# 返回构建结果
Return('LIBPATH', 'LIBS')

File: components/rust/rtt_rust-v1.0.0/rtt_main/Sconscript.c
Import('rtconfig')
from building import *
import os

cwd = GetCurrentDir()

# 添加Rust应用支持
if GetDepend(['PKG_USING_RTT_RUST']):
    # 添加applications目录下的Rust应用
    rust_apps = []
    apps_dir = os.path.join(cwd, 'applications')
    
    if os.path.exists(apps_dir):
        for app_name in os.listdir(apps_dir):
            app_path = os.path.join(apps_dir, app_name)
            sconscript_path = os.path.join(app_path, 'SConscript')
            
            if os.path.isdir(app_path) and os.path.exists(sconscript_path):
                print("Found Rust application: {}".format(app_name))
                rust_apps.append(app_path)
    
    # 构建每个Rust应用
    group = []
    for app_path in rust_apps:
        try:
            result = SConscript(os.path.join(app_path, 'SConscript'))
            if result:
                if isinstance(result, tuple) and len(result) >= 2:
                    LIBPATH, LIBS = result[:2]
                    group.append({'LIBPATH': LIBPATH, 'LIBS': LIBS})
                print("Successfully built Rust app in: {}".format(app_path))
        except Exception as e:
            print("Failed to build Rust app in {}: {}".format(app_path, str(e)))
    
    Return('group')

Return([])